Slim/Control/Queries.pm

```diff
@@ -4636,26 +4689,41 @@ sub worksQuery {
 		my $sth = $dbh->prepare_cached($sql);
 		$sth->execute( @{$p} );
 
-		my ($work, $workId, $composer, $composerId, $nameSort, $titleSort );
-		$sth->bind_columns(\$work, \$workId, \$composer, \$composerId, \$nameSort, \$titleSort);
+		my ($work, $workId, $composer, $composerId, $nameSort, $titleSort, $covers, $album_ids);
+		$sth->bind_columns(\$work, \$workId, \$composer, \$composerId, \$nameSort, \$titleSort, \$covers, \$album_ids);
 
 		while ( $sth->fetch ) {
-			#$id += 0;
+
+			my @allImages = split(/,/,$covers);
+			my $image = @allImages[0];
+			my $images = scalar @allImages > 4 ? join(',', @allImages[0..3]) : join(',',@allImages);
+			
+
 			utf8::decode($work) if $work;
 			utf8::decode($composer) if $composer;
-			$request->addResultLoop($loopname, $chunkCount, 'work', $work);
 			$request->addResultLoop($loopname, $chunkCount, 'work_id', $workId);
-			$request->addResultLoop($loopname, $chunkCount, 'composer', $composer);
+			if ( $composerCount == 1 && $composerId eq $artistID ) {
+				$request->addResultLoop($loopname, $chunkCount, 'composer', $work);
+				$request->addResultLoop($loopname, $chunkCount, 'work', " ");
+			} else {
+				$request->addResultLoop($loopname, $chunkCount, 'composer', $composer);
+				$request->addResultLoop($loopname, $chunkCount, 'work', $work);
+			}
 			$request->addResultLoop($loopname, $chunkCount, 'composer_id', $composerId);
 			$request->addResultLoop($loopname, $chunkCount, 'image', "html/images/works.png");
-#			$request->addResultLoop($loopname, $chunkCount, 'year', $year);
-#			$request->addResultLoop($loopname, $chunkCount, 'from_search', $search) if $search;
+			$request->addResultLoop($loopname, $chunkCount, 'image', $image);
+			$request->addResultLoop($loopname, $chunkCount, 'images', $images);
+			$request->addResultLoop($loopname, $chunkCount, 'album_id', $album_ids);
 
 			my $textKey;
 			utf8::decode( $nameSort ) if $nameSort;
-			$textKey = substr $nameSort, 0, 1;
+			$textKey = $composerCount == 1 && $composerId eq $artistID ? substr $titleSort, 0, 1 : substr $nameSort, 0, 1;
```

Comment:
Similarly to a previous comment: please only wrap the variable decision (`$titlesort` vs `namesort`) inside the decision, but not the `substr` cal. And please use brackets with function calls, even if Perl allows to call them without (`substr($var, 0, 1)`).